lib/iris/tests/unit/lazy_data/test_array_masked_to_nans.py

```diff
@@ -31,41 +31,45 @@
 
 
 class Test(tests.IrisTest):
-    def test_masked(self):
-        masked_array = ma.masked_array([[1.0, 2.0], [3.0, 4.0]],
-                                       mask=[[0, 1], [0, 0]])
-
-        result = array_masked_to_nans(masked_array).data
-
+    def _common_checks(self, result):
         self.assertIsInstance(result, np.ndarray)
         self.assertFalse(isinstance(result, ma.MaskedArray))
         self.assertFalse(ma.is_masked(result))
 
+    def test_masked_input(self):
+        masked_array = ma.masked_array([[1.0, 2.0], [3.0, 4.0]],
+                                       mask=[[0, 1], [0, 0]])
+
+        result = array_masked_to_nans(masked_array)
+
+        self._common_checks(result)
         self.assertArrayAllClose(np.isnan(result),
                                  [[False, True], [False, False]])
         result[0, 1] = 777.7
         self.assertArrayAllClose(result, [[1.0, 777.7], [3.0, 4.0]])
 
+    def test_unmasked_input(self):
+        unmasked_array = np.array([1.0, 2.0])
+        result = array_masked_to_nans(unmasked_array)
+        # Non-masked array is returned as-is, without copying.
+        self.assertIs(result, unmasked_array)
+
     def test_empty_mask(self):
         masked_array = ma.masked_array([1.0, 2.0], mask=[0, 0])
 
-        result = array_masked_to_nans(masked_array).data
-
-        self.assertIsInstance(result, np.ndarray)
-        self.assertFalse(isinstance(result, ma.MaskedArray))
-        self.assertFalse(ma.is_masked(result))
+        result = array_masked_to_nans(masked_array)
 
-        # self.assertIs(result, masked_array.data)
-        # NOTE: Wanted to check that result in this case is delivered without
-        # copying.  However, it seems that ".data" is not just an internal
-        # reference, so copying *does* occur in this case.
+        self._common_checks(result)
         self.assertArrayAllClose(result, masked_array.data)
 
-    def test_non_masked(self):
-        unmasked_array = np.array([1.0, 2.0])
-        result = array_masked_to_nans(unmasked_array, mask=False)
-        # Non-masked array is returned as-is, without copying.
-        self.assertIs(result, unmasked_array)
+    def test_integer_array(self):
```

Comment:
This tests "masked integer" input.
We should have another one that tests "un-masked integers", where the input is a non-masked integer array, as in that case the dtype is preserved.
Also I suppose yet another, to test an integer masked array with no masked points (like 'test_empty_mask' above).